### ISSUE TITLE
Object classification uses Sigmoid+binary_cross_entropy, why not keep same as DETR using Softmax+cross_entropy?

### ISSUE THREAD

**Issue by gitzyong812** [fundamentalvision/Deformable-DETR#72]
Thanks for sharing the code. I am wondering why not using Softmax+cross_entropy?
Did you see any differences in performance?

**Comment by ifzhang** [comment]
@zyong812 The difference may lie in the background class. Sigmoid + BCE does not need an extra class for the background. The background class is just an all-zero one-hot vector. Softmax + CE needs an extra class for the background. So the total class in DETR is num_classes + 1 and the total class in Deformable DETR is num_classes. The performance between the two methods may be similar.

**Comment by d12306** [comment]
@ifzhang , @zyong812 , have you guys tried out using the cross entropy loss for training the deformable detr? i actually get worse performance under the same training epochs

**Comment by ldtho** [comment]
@d12306 , which one is worse? Sigmoid + BCE or Softmax + CE

**Comment by jiangxiluning** [comment]
Softmax + CE

**Comment by theneotopia** [comment]
> @zyong812 The difference may lie in the background class. Sigmoid + BCE does not need an extra class for the background. The background class is just an all-zero one-hot vector. Softmax + CE needs an extra class for the background. So the total class in DETR is num_classes + 1 and the total class in Deformable DETR is num_classes. The performance between the two methods may be similar.

The code seems to choose the top-k predictions from all binary predictions, wouldn't this incur that a query (predicted bounding box) can be assigned to multiple classes if this query produces many high sigmoid scores.
`topk_values, topk_indexes = torch.topk(prob.view(out_logits.shape[0], -1), 100, dim=1)`

**Comment by nikhilchh** [comment]
So if I have 1 class (say humans) then should I set num_classes=1 ?
